slatedb/src/db/builder.rs

```diff
@@ -258,7 +284,8 @@ impl<P: Into<Path>> DbBuilder<P> {
         // Create path resolver and table store
         let path_resolver = PathResolver::new_with_external_ssts(path.clone(), external_ssts);
         let table_store = Arc::new(TableStore::new_with_fp_registry(
-            maybe_cached_object_store.clone(),
+            maybe_cached_main_object_store.clone(),
```

Comment:
Not for this PR, but I feel like it's probably worth stamping the object store that we use with some marker object that designates it as a wal/main object store and then checking that when initializing tablestore. It feels too easy to mix these up by mistake.

Reply:
Alternatively we could just pass the manifest and main object store (when it has the URIs) and initialize the object store Arcs that way. This ties in to my earlier comment - we can have some wrapping type that encapsulates the different object stores and then initialize that type using the main object store plus the manifest.